tools/ampir/ampir.xml

```diff
@@ -0,0 +1,87 @@
+<tool name="ampir" id="ampir" version="1.1.0">
```

Comment:
You definitely want a profile version here

Reply:
I fixed it when @bgruening mentioned it but now I see it was in the wrong place :( Currently rebuilding everything.

Reply:
```
<tool profile="22.05" name="yara_mapper" id="yara_mapper" version="1.17">
  <!--Source in git at: https://github.com/fubar2/galaxy_tf_overlay-->
  <!--Created by toolfactory@galaxy.org at 04/09/2023 18:31:39 using the Galaxy Tool Factory.-->
```
now and for any future generated tool.

Reply:
@mvdbeek @bgruening - I think these are done now.
Thanks for reviewing!

Reply:
I only commented on the obvious stuff. You want to drop the stdio section too, drop the unused script, write a help section and address @bgruening's comments. I don't think the auto-generation should lower the usual threshold we have for IUC tools.

Reply:
tools/openms/FidoAdapter.xml:11:  <expand macro="stdio"/>
tools/openms/RTEvaluation.xml:11:  <expand macro="stdio"/>
tools/openms/FeatureFinderMetabo.xml:11:  <expand macro="stdio"/>
tools/openms/OpenSwathDecoyGenerator.xml:11:  <expand macro="stdio"/>
tools/openms/FFEval.xml:11:  <expand macro="stdio"/>
tools/openms/OpenSwathWorkflow.xml:11:  <expand macro="stdio"/>
tools/openms/PTPredict.xml:11:  <expand macro="stdio"/>
tools/openms/ExternalCalibration.xml:11:  <expand macro="stdio"/>
tools/openms/CVInspector.xml:11:  <expand macro="stdio"/>
tools/openms/XTandemAdapter.xml:11:  <expand macro="stdio"/>
tools/openms/PhosphoScoring.xml:11:  <expand macro="stdio"/>
tools/openms/HighResPrecursorMassCorrector.xml:11:  <expand macro="stdio"/>
tools/openms/PeptideIndexer.xml:11:  <expand macro="stdio"/>
tools/openms/OpenSwathFileSplitter.xml:11:  <expand macro="stdio"/>
tools/openms/PeakPickerWavelet.xml:11:  <expand macro="stdio"/>
tools/openms/MapRTTransformer.xml:11:  <expand macro="stdio"/>
tools/openms/QCMerger.xml:11:  <expand macro="stdio"/>
tools/openms/MRMMapper.xml:11:  <expand macro="stdio"/>
tools/openms/SpectraMerger.xml:11:  <expand macro="stdio"/>
tools/openms/XMLValidator.xml:11:  <expand macro="stdio"/>
tools/openms/FeatureLinkerLabeled.xml:11:  <expand macro="stdio"/>
tools/openms/TransformationEvaluation.xml:11:  <expand macro="stdio"/>
tools/numeric_clustering/numeric_clustering.xml:6:    <stdio>
tools/augustus/augustus.xml:12:    </stdio>
tools/crt/crt.xml:8:    <stdio>
tools/quast/quast.xml:28:    </stdio>
tools/metilene/metilene.xml:12:    <stdio>
tools/metilene/metilene.xml:15:    </stdio>
tools/text_processing/text_processing/macros.xml:9:    <xml name="stdio">
tools/minced/minced.xml:7:    <stdio>
tools/minced/minced.xml:10:    </stdio>
tools/format_cd_hit_output/format_cd_hit_output.xml:7:    <stdio>
tools/format_cd_hit_output/format_cd_hit_output.xml:9:    </stdio>
tools/rna_tools/miRDeep2/miRDeep2_quantifier/quantifier.xml:6:    <stdio>
tools/rna_tools/miRDeep2/miRDeep2_quantifier/quantifier.xml:13:    </stdio>
tools/rna_tools/cmv/cmcv.xml:7:    <expand macro="stdio" />
tools/rna_tools/dotknot/dotknot.xml:12:    </stdio>
tools/rna_tools/infernal/cmstat.xml:7:    <expand macro="stdio" />
tools/rna_tools/locarna/locarna_exparnap.xml:12:    <expand macro="stdio" />
tools/rna_tools/vienna_rna/rnaplex.xml:9:    <expand macro="stdio" />
tools/rna_tools/vienna_rna/rnaplot.xml:8:    <expand macro="stdio" />
tools/rna_tools/ribotaper/ribotaper_part2_create_metaplots.xml:7:    <stdio>
tools/instagraal/instagraal.xml:22:    </stdio>
tools/GraphClust/Plotting/MotifFinderPlot.xml:7:  <stdio>
tools/GraphClust/Plotting/MotifFinderPlot.xml:9:  </stdio>
tools/GraphClust/PrepareForMlocarna/preMlocarna.xml:6:  <stdio>
tools/GraphClust/PrepareForMlocarna/preMlocarna.xml:8:  </stdio>
tools/GraphClust/NSPDK/NSPDK_sparseVect.xml:6:    <stdio>
tools/GraphClust/NSPDK/NSPDK_candidateClusters.xml:8:    </stdio>
tools/GraphClust/GSPAN/fasta2shrep_gspan.xml:6:    <stdio>
tools/GraphClust/GSPAN/fasta2shrep_gspan.xml:8:    </stdio>
tools/GraphClust/LocARNAGraphClust/locarna_best_subtree.xml:8:  <stdio>
tools/GraphClust/LocARNAGraphClust/locarna_best_subtree.xml:10:  </stdio>
tools/GraphClust/Preprocessing/preprocessing.xml:8:    <stdio>
tools/GraphClust/Preprocessing/preprocessing.xml:10:    </stdio>
tools/jamm/jamm.xml:28:<stdio>
tools/jamm/jamm.xml:32:</stdio>
tools/platypus/platypus.xml:6:    <stdio>
tools/image_processing/imagej2/imagej2_macros.xml:10:        <stdio>
tools/image_processing/imagej2/imagej2_macros.xml:15:        </stdio>
tools/EDeN/EDeN/EDeN_feature.xml:36:    <stdio>
tools/EDeN/EDeN/EDeN_feature.xml:41:    </stdio>
tools/twistdna/twistdna.xml:6:    <stdio>
tools/twistdna/twistdna.xml:9:    </stdio>
tools/protease_prediction/protease.xml:7:    <expand macro="stdio"/>
tools/protease_prediction/macros.xml:8:    <xml name="stdio">
tools/protease_prediction/macros.xml:9:        <stdio>
tools/protease_prediction/macros.xml:11:        </stdio>
tools/labels/labels.xml:8:    <stdio>
tools/labels/labels.xml:13:    </stdio>
tools/ampir/ampir.xml:8:  <stdio>
tools/ampir/ampir.xml:12:  </stdio>
tools/sqlite_merger/sqlite_merger.xml:19:    <stdio>
tools/sqlite_merger/sqlite_merger.xml:24:    </stdio>
tools/mafft/mafft-add.xml:8:  <stdio>
tools/vt/vt_macros.xml:16:    <xml name="stdio">
tools/vt/vt_macros.xml:17:        <stdio>
tools/vt/vt_macros.xml:22:        </stdio>
tools/vt/vt_normalize.xml:8:    <expand macro="stdio" />
tools/vt/vt_decompose.xml:8:    <expand macro="stdio" />
tools/bigwig_to_bedgraph/bigwig_to_bedgraph.xml:8:    <stdio>
tools/bigwig_to_bedgraph/bigwig_to_bedgraph.xml:10:    </stdio>
tools/sklearn/pre_process.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/feature_selection.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/to_categorical.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/model_prediction.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/discriminant.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/keras_macros.xml:4:  <xml name="macro_stdio">
tools/sklearn/keras_macros.xml:5:    <stdio>
tools/sklearn/keras_macros.xml:7:    </stdio>
tools/sklearn/model_validation.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/regression_metrics.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/pca.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/label_encoder.xml:7:    <expand macro="macro_stdio"/>
tools/sklearn/nn_classifier.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/stacking_ensembles.xml:41:    <expand macro="macro_stdio" />
tools/sklearn/fitted_model_eval.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/main_macros.xml:13:    <xml name="macro_stdio">
tools/sklearn/main_macros.xml:14:    <stdio>
tools/sklearn/main_macros.xml:18:    </stdio>
tools/sklearn/numeric_clustering.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/lightgbm.xml:9:    <expand macro="macro_stdio" />
tools/sklearn/simple_model_fit.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/association_rules.xml:7:    <expand macro="macro_stdio"/>
tools/sklearn/train_test_eval.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/svm.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/estimator_attributes.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/pairwise_metrics.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/train_test_split.xml:12:    <expand macro="macro_stdio" />
tools/sklearn/pipeline.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/keras_train_and_eval.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/search_model_validation.xml:32:    <expand macro="macro_stdio" />
tools/sklearn/generalized_linear.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/clf_metrics.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/keras_model_builder.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/sparse.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/ml_visualization_ex.xml:8:    <expand macro="macro_stdio" />
tools/sklearn/keras_model_config.xml:8:  <expand macro="macro_stdio" />
tools/sklearn/sample_generator.xml:7:    <expand macro="macro_stdio" />
tools/sklearn/keras_batch_models.xml:8:  <expand macro="macro_stdio" />
tools/sklearn/ensemble.xml:7:    <expand macro="macro_stdio" />
tools/diff/diff.xml:11:    <stdio>
tools/diff/diff.xml:14:    </stdio>
tools/format_metaphlan2_output/format_metaphlan2_output.xml:7:    <stdio>
tools/format_metaphlan2_output/format_metaphlan2_output.xml:8:    </stdio>
data_managers/data_manager_sortmerna_database_downloader/data_manager/data_manager_sortmerna_download.xml:10:    <stdio>
data_managers/data_manager_sortmerna_database_downloader/data_manager/data_manager_sortmerna_download.xml:13:    </stdio>
chemicaltoolbox/openbabel/macros.xml:33:    <xml name="stdio">
chemicaltoolbox/openbabel/macros.xml:34:        <stdio>
chemicaltoolbox/openbabel/macros.xml:36:        </stdio>
chemicaltoolbox/opsin/opsin.xml:7:    <stdio>
chemicaltoolbox/opsin/opsin.xml:12:    </stdio>
chemicaltoolbox/data_source/get_pubchem/get_pubchem_as_smiles.xml:7:    <stdio>
chemicaltoolbox/data_source/get_pubchem/get_pubchem_as_smiles.xml:20:    </stdio>
chemicaltoolbox/autodock_vina/prepare_ligand/prepare_ligand.xml:10:    <stdio>
chemicaltoolbox/autodock_vina/prepare_ligand/prepare_ligand.xml:12:    </stdio>
chemicaltoolbox/autodock_vina/prepare_receptor/prepare_receptor.xml:10:    <stdio>
chemicaltoolbox/autodock_vina/prepare_receptor/prepare_receptor.xml:12:    </stdio>

Reply:
@mvdbeek https://github.com/hexylena/galaxyxml/pull/38 fixes it I think as far as is feasible adding detect_errors="aggressive" to the command tag. I copied that stdio from one of Bjoern's recent tools and see that most stull use it but am happy to lead the charge once the PR is in. Automated help is a stretch since the nf-core stuff has none to speak of, but the hand-crafted ones, panaroo and yara have handcrafted and I hope quite reasonable help from their documentation. Ampir is completely automated from the nf-core module and test. A different order of not IUC.